### PR TITLE
Trn 300

### PR DESCRIPTION
**Pull request by pchristopher1275**
https://jira2.iparadigms.com/browse/TRN-300 

The crux of the solution:

* Add 2 fields: last_dispatch (timestamp) and last_empty_dispatch (timestamp)
* Update last_dispatch every time we finish a dispatch with >0 links, and we update last_empty_dispatch every time it's 0 links.
* Use a config parameter empty_dispatch_retry_interval (similar to full_scan_period)
* When considering a domain, If last_empty_dispatch > last_dispatch, then check if it's been long enough to try dispatching again (using empty_dispatch_retry_interval)

### CODE REVIEW

File: cassandra/dispatcher.go [hunk 1]
```diff
@@ -463,6 +471,25 @@ func (d *Dispatcher) generateSegment(domain string) error {
...
+	if err != nil {
+		log4go.Error("Failed to read last_dispatch and last_empty_dispatch for %q: %v", domain, err)
+		return err
+	}
+	if lastEmptyDispatch.After(lastDispatch) && time.Since(lastEmptyDispatch) < d.emptyDispatchRetryInterval {
+		err := d.db.Query(`UPDATE domain_info SET dispatched = false WHERE dom = ?`, domain).Exec()
+		if err != nil {
+			return fmt.Errorf("Premature return: error resetting domain %q: %v", domain, err)
+		}
+		return nil
```
Comment (dankinder) [inline cassandra/dispatcher.go]: Shouldn't `dispatched` already be false for this domain? Seems like all we need in this condition is `return`.'

One other quick request here: could we move the "Generating a crawl segment for..." log statement below this block, and maybe add a DEBUG line if we're skipping it to indicate that it's being skipped?

Comment (pchristopher1275) [inline cassandra/dispatcher.go]: Yes, you are correct. And yes, I'll make those changes.

**Comment by pchristopher1275** [comment]
BUMP: This patch has been sitting here for a while. AFAIK, it's ready to be merged (or rejected).

**Comment by dankinder** [comment]
:facepunch:

**Pull request merged**